### PR TITLE
Change the property names to be lower case letter in the tspCodeModel.json

### LINKED ISSUE #4715: Properties of models in TCGC starts with lower case letter, but ours start with upper case letters
I think it is fine that we deserialize the `tspCodeModel.json` file into our types with property names **case-insensitively**

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes #4715
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDateTimeTypeConverter.cs [hunk 1]
```diff
@@ -17,40 +17,43 @@ public TypeSpecInputDateTimeTypeConverter(TypeSpecReferenceHandler referenceHand
...
                 }
             }
 
-            if (type is not InputPrimitiveType wireType)
-            {
-                throw new JsonException("The wireType of a DateTime type must be a primitive type");
-            }
-
+            name = name ?? throw new JsonException("DateTime type must have name");
+            crossLanguagueDefinitionId= crossLanguagueDefinitionId?? throw new JsonException("DateTime type must have crossLanguagueDefinitionId");
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDateTimeTypeConverter.cs:51]:
```suggestion
            crossLanguagueDefinitionId= crossLanguagueDefinitionId ?? throw new JsonException("DateTime type must have crossLanguagueDefinitionId");
```

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputLiteralTypeConverter.cs [hunk 2]
```diff
@@ -25,79 +25,56 @@ public override void Write(Utf8JsonWriter writer, InputLiteralType value, JsonSe
...
+            JsonElement? rawValue = null;
+            InputType? valueType = null;
             IReadOnlyList<InputDecoratorInfo>? decorators = null;
 
             while (reader.TokenType != JsonTokenType.EndObject)
             {
                 var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
-                    || reader.TryReadWithConverter(nameof(InputLiteralType.ValueType), options, ref type)
-                    || reader.TryReadWithConverter(nameof(InputLiteralType.Decorators), options, ref decorators);
+                    || reader.TryReadWithConverter("value", options, ref rawValue)
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputLiteralTypeConverter.cs:35]: which converter is this applied with?

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs [hunk 3]
```diff
@@ -23,19 +23,21 @@ public override InputEnumTypeValue Read(ref Utf8JsonReader reader, Type typeToCo
...
             while (reader.TokenType != JsonTokenType.EndObject)
             {
                 var isKnownProperty = reader.TryReadReferenceId(ref isFirstProperty, ref id)
-                    || reader.TryReadString(nameof(InputEnumTypeValue.Name), ref name)
-                    || reader.TryReadEnumValue(nameof(InputEnumTypeValue.Value), ref value)
-                    || reader.TryReadString(nameof(InputEnumTypeValue.Description), ref description)
-                    || reader.TryReadWithConverter(nameof(InputEnumTypeValue.Decorators), options, ref decorators);
+                    || reader.TryReadString("name", ref name)
+                    || reader.TryReadWithConverter("value", options, ref rawValue)
+                    || reader.TryReadWithConverter("valueType", options, ref valueType)
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs:38]: is this valueType used anywhere else apart from this deserialization? If not, do we need it in the InputEnumTypeValue?

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDateTimeTypeConverter.cs (hunk 1 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputDateTimeTypeConverter.cs:51]: Fixed

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs (hunk 3 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs:38]: we used it below to determine which concrete type we want to use for the raw value (whose type is `object`)

Before this PR, we are using the `ValueType` from the enum therefore there is a lot of workarounds, which are all removed now

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputLiteralTypeConverter.cs (hunk 2 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputLiteralTypeConverter.cs:35]: every type has a default converter if not specified.
This is invoking the default converter for `JsonElement`.
We could do this for all types.

File: src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs [hunk 4]
```diff
@@ -45,11 +47,19 @@ public static InputEnumTypeValue CreateEnumTypeValue(ref Utf8JsonReader reader,
...
+            }
+
+            valueType = valueType ?? throw new JsonException("EnumValue must have valueType");
 
-            var enumValue = new InputEnumTypeValue(name, value, description)
+            InputEnumTypeValue enumValue = valueType.Kind switch
             {
-                Decorators = decorators ?? Array.Empty<InputDecoratorInfo>(),
+                InputPrimitiveTypeKind.String => new InputEnumTypeStringValue(name, rawValue.Value.GetString() ?? throw new JsonException(), description) { Decorators = decorators ?? [] },
+                InputPrimitiveTypeKind.Int32 => new InputEnumTypeIntegerValue(name, rawValue.Value.GetInt32(), valueType, description) { Decorators = decorators ?? [] },
```
Comment (live1206) [inline src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputEnumTypeValueConverter.cs:60]: valueType should be removed here

**Review by live1206 — APPROVED** [review]

**Pull request merged**